### PR TITLE
Add EOL check to build to prevent unintended mixed/crlf changes

### PR DESCRIPTION
**Pull request by tex3d**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -113,6 +113,24 @@ if ( "$ENV{CI}" AND "$ENV{APPVEYOR}" )
...
+    COMMENT Running hcteolcheck.cmd - fails if i/crlf or i/mixed in index
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMAND ${HCTEOLCHECK}
+    VERBATIM)
+else()
+  add_custom_target(git_eol_check
+    ALL
+    COMMENT Running EOL check - fails if i/crlf or i/mixed in index
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMAND ! git ls-files --eol | grep -e "i/crlf" -e "i/mixed"
```
Comment (llvm-beanz) [inline CMakeLists.txt:129]: We should probably run this as `sh -c '! git ls-files --eol | grep -e "i/crlf" -e "i/mixed"'`, that way weirdos like me who have non-posix compliant default shells won't get broken by this :)

**Review by llvm-beanz — COMMENTED** [review]
One small change, otherwise looks good.

File: CMakeLists.txt (hunk 1 above)
Comment (tex3d) [reply CMakeLists.txt:129]: Great suggestion, thanks!

**Review by llvm-beanz — APPROVED** [review]
LGTM

**Review by raghavthind2005 — APPROVED** [review]

**Comment by llvm-beanz** [comment]
@tex3d, do you want to rebase and merge this? If not, please close the PR.

**Pull request closed without merging**